### PR TITLE
fix(files): Fix navigating with breadcrumbs not highlighting navigation entry

### PR DESCRIPTION
**Pull request by Pytal**
## Summary

Before | After
--- | ---
image | image

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [ ] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [x] Screenshots before/after for front-end changes
- [x] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [ ] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

File: apps/files/src/components/BreadCrumbs.vue [hunk 1]
```diff
@@ -171,8 +172,10 @@ export default defineComponent({
 				}
 			}
 			if (node === undefined) {
+				const view = this.views.find(view => view.params?.dir === dir)
 				return {
 					...this.$route,
+					params: { fileid: String(view?.params?.fileid) },
```
Comment (susnux) [inline apps/files/src/components/BreadCrumbs.vue:178]: If this is undefined then fileid will be the string `undefined`, no? So maybe better:
```suggestion
					params: { fileid: String(view?.params?.fileid ?? '') },
```

Comment (susnux) [inline apps/files/src/components/BreadCrumbs.vue:178]: If this is undefined then fileid will be the string `'undefined'`, no?

Comment (Pytal) [reply apps/files/src/components/BreadCrumbs.vue]: Yes though it does not break anything routing-wise as the fileid param matches numbers only

**Review by susnux — APPROVED** [review]

**Review by artonge — APPROVED** [review]

**Comment by Pytal** [comment]
/backport 930a5141ea9a34e0624a2d495e0f2708f7593e03 to stable30

**Pull request merged**